### PR TITLE
Changes for Rosetta API

### PR DESCRIPTION
**Pull request by syuan100**
- Implicit burn endpoint
- Peers endpoint
- Snapshot + core bump
- Dev implicit_burn config
- Rosetta Docker config

### CODE REVIEW

**Comment by syuan100** [comment]
@abhay This is the final piece of the puzzle for the Rosetta API, and then we'll have no more forked deps. Let me know what y'all think

**Comment by abhay** [comment]
> @abhay This is the final piece of the puzzle for the Rosetta API, and then we'll have no more forked deps. Let me know what y'all think

Is the peers RPC required in this implementation?

**Comment by abhay** [comment]
@madninja curious on your thoughts of always storing implicit burns in `blockchain-node`? Seems useful for wallet / light nodes.

**Comment by madninja** [comment]
> @madninja curious on your thoughts of always storing implicit burns in `blockchain-node`? Seems useful for wallet / light nodes.

We need this for both etl and node actually.. yes

My initial thinking though was to include the implicit burn in the transaction json itself as well

**Comment by syuan100** [comment]
> > @abhay This is the final piece of the puzzle for the Rosetta API, and then we'll have no more forked deps. Let me know what y'all think
> 
> Is the peers RPC required in this implementation?

For the `/network/status` endpoint it is. I just included the bulk of the implementation from miner intact in case it might be useful in the future. I can pare it down to only what's necessary though.

**Comment by madninja** [comment]
> > > @abhay This is the final piece of the puzzle for the Rosetta API, and then we'll have no more forked deps. Let me know what y'all think
> > 
> > 
> > Is the peers RPC required in this implementation?
> 
> For the `/network/status` endpoint it is. I just included the bulk of the implementation from miner intact in case it might be useful in the future. I can pare it down to only what's necessary though.

Who is that useful for though? If it's just for helpful debugging maybe don't include it in this PR?

**Comment by syuan100** [comment]
Sure I'll update later this afternoon to cleanup the peers endpoint

**Comment by syuan100** [comment]
> > @madninja curious on your thoughts of always storing implicit burns in `blockchain-node`? Seems useful for wallet / light nodes.
> 
> We need this for both etl and node actually.. yes
> 
> My initial thinking though was to include the implicit burn in the transaction json itself as well

I'll take a stab and add it as a commit.

**Comment by syuan100** [comment]
@abhay @madninja updated per the notes above. Tested locally and it works properly, however there is one codepath I'm not able to test which is: https://github.com/helium/blockchain-node/blob/master/src/bn_txns.erl#L133

Essentially, it looks like there is a JsonCF where txn data lives, but in the case that it's not there we go grab the actual txn and then parse it into json. I haven't been able to find a transaction that isn't found in the JsonCF so wanted to ask to see how I might be able to trigger it.

**Comment by madninja** [comment]
> Essentially, it looks like there is a JsonCF where txn data lives, but in the case that it's not there we go grab the actual txn and then parse it into json. I haven't been able to find a transaction that isn't found in the JsonCF so wanted to ask to see how I might be able to trigger it.

That codepath was for blockchain-node instances that were running _before_ the jsoncf column family was added. It ensures that if jsoncf doesn't have it but the transaction exists that it is still returned as correctly as we can

**Comment by abhay** [comment]
> @abhay @madninja updated per the notes above. Tested locally and it works properly

We can probably remove that new overlay config now too right?

**Comment by syuan100** [comment]
> > @abhay @madninja updated per the notes above. Tested locally and it works properly
> 
> We can probably remove that new overlay config now too right?

Sooo rosetta requires the data dir to be at `/data`... so while it's only a one line change I still need the docker rosetta config. devib has been removed though.

**Comment by syuan100** [comment]
> That codepath was for blockchain-node instances that were running _before_ the jsoncf column family was added. It ensures that if jsoncf doesn't have it but the transaction exists that it is still returned as correctly as we can

ok I was able to test it just by commenting out the JsonCF codepath and forcing the node to pull the actual txn. Works accordingly

**Comment by syuan100** [comment]
Hey Helium team, I also included new testnet configs on a separate branch since I want to do more rigorous testing for more types of transactions. I'm wondering if I should just add it into this branch as part of the PR?

**Comment by syuan100** [comment]
I had based it off this branch. Link for reference:
https://github.com/syuan100/blockchain-node/tree/testnet-configs

**Comment by syuan100** [comment]
Hi there, I decided to just add the testnet-configs into this PR, as it will help with HTLC transaction testing.

Could possibly be landed before the Wed dewi call? It would be good to have something non-forked in the dockerfile. All good if not, just thought I should ask though.

**Comment by abhay** [comment]
> Could possibly be landed before the Wed dewi call? It would be good to have something non-forked in the dockerfile. All good if not, just thought I should ask though.

Sorry I missed this. If @madninja is good with it, I'm a 👍🏼

File: config/sys.config [hunk 1]
```diff
@@ -28,10 +28,10 @@
 
         {honor_quick_sync, true},
         {quick_sync_mode, blessed_snapshot},
-        {blessed_snapshot_block_height, 959041},
+        {blessed_snapshot_block_height, 975601},
```
Comment (abhay) [inline config/sys.config]:
```suggestion
        {blessed_snapshot_block_height, 983521},
```

File: config/sys.config [hunk 2]
```diff
@@ -28,10 +28,10 @@
...
         {blessed_snapshot_block_hash,
-            <<34, 206, 154, 249, 18, 46, 227, 229, 192, 100, 101, 102, 243, 107, 119, 207, 57, 219,
-                106, 30, 231, 29, 82, 221, 234, 116, 189, 64, 1, 70, 55, 143>>},
+            <<87,86,149,234,13,212,196,250,56,57,160,76,150,198,152,134,4,189,17,109,213,98,182,6,194,119,59,127,120,219,229,108>>},
```
Comment (abhay) [inline config/sys.config]:
```suggestion
<<97,208,157,201,5,40,83,139,102,73,148,246,147,130,211,30,45,143,70,112,163,70,105,238,101,244,197,179,72,14,100,15>>},
```

**Review by abhay — CHANGES_REQUESTED** [review]
updated snapshot to match miner master

**Comment by madninja** [comment]
Would you mind rebasing on master one more time @syuan100 ? I'll should be able to get it merged this week

**Comment by syuan100** [comment]
Rebased and tested, works as expected with latest changes from master

**Comment by abhay** [comment]
@madninja is this good to go?

**Review by madninja — APPROVED** [review]

**Pull request merged**